src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingPassthrough.scala

```diff
@@ -63,10 +63,11 @@ trait StreamingPassthrough
     * The passthrough implementation:
     *   - `mapRequest` to transform the incoming request to what we want to send to the remote system
     *   - `extractRequest` so we have the transformed request as an object
+   *    - `remotePathOverride` to provide a pre-configured path if remote path structure is different from local
     *   - call the remote system and reply to the user via `routeResponse` streaming
    */
-  private def passthroughImpl(localBasePath: Uri.Path, remoteBaseUri: Uri): Route = {
-    mapRequest(transformToPassthroughRequest(localBasePath, remoteBaseUri)) {
+  def passthroughImpl(localBasePath: Uri.Path, remoteBaseUri: Uri, remotePathOverride: Option[String] = None): Route = {
```

Comment:
We could make a new function like `streamingRedirect(passthroughMapping: (Uri.Path, Uri))` that is clear about the fact that it sends the request to a different path, rather than just passing through.

Then, implement its body with with a call to the private `passthroughImpl` that supports `remotePathOverride`.

Finally, replace
```
passthroughImpl(Uri.Path(localBase), Uri(rawlsWorkflowRoot), Option(s"/${workflowId}/genomics/${suffix}"))
```
with
```
streamingRedirect(Uri.Path(localBase) -> Uri(s"$rawlsWorkflowRoot/${workflowId}/genomics/${suffix}"))
```

(The thought that sent me down this path was that functions with `impl` in the name are usually intended to be the hidden implementation behind a more ergonomic public facade.)

Reply:
Wasn't aware of the `impl` naming convention, that's good to know. It lines up with why the method was defined `private` to begin with.